llvm/lib/Target/AIE/AIEHazardRecognizer.cpp

```diff
@@ -454,32 +480,45 @@ bool AIEHazardRecognizer::checkConflict(
 }
 
 void AIEHazardRecognizer::emitInScoreboard(const MCInstrDesc &Desc,
+                                           MemoryBankBits MemoryBanks,
                                            int DeltaCycles) {
   enterResources(
       Scoreboard, ItinData, Desc.getSchedClass(),
       getSlotSet(Desc, *TII->getFormatInterface(), IgnoreUnknownSlotSets),
-      DeltaCycles, FUDepthLimit);
+      MemoryBanks, TII->getFirstMemoryCycle(Desc.getSchedClass()), DeltaCycles,
```

Comment:
A- What about LastMemoryCycle?
B- I think a load and a store don't conflict in the same bank.

Reply:
I will think more about part (A) and how to handle that, 
w.r.t to part (B) we **do** have conflict in the same bank for load & store.

Reply:
Correct me if I'm wrong: Realistically, we only care about bank conflicts for loads. We have only one store unit, so we do not need to track parallel write accesses. If we have a parallel load and store to the same bank, I would assume this is fine as well, as we have distinct read and write ports. This is only problematic if those touch the same location, but if that happens, we have bigger problems and our AA does not work.

So I think it's fine to only care about `TII->getFirstMemoryCycle()` because all the loads happen in the same cycle (at least for AIE2). We can add a comment in the code to explain this, and maybe an assert like `assert(TII->getFirstMemoryCycle(...) == TII->getMinFirstMemoryCycle());`

Please share your thoughts @martien-de-jong @krishnamtibrewala :)

Reply:
We cannot do LOAD & STORE to same bank that will lead to bank conflict and will trigger HW stall, have confirmed this with ARCH team and personally verified it by understanding the simulation model of AIE2.